scripts/upgrade-npm.js

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env node
+
+/**
+ * This upgrades npm on Travis when using Node.js 0.8x
+ */
+
+if (process.env.TRAVIS_NODE_VERSION === '0.8') {
+  var exec = require('child_process').exec;
+  // ensure *dependencies* are installed using provided npm
+  exec('npm install --production', function (err) {
+    if (err) {
+      throw new Error(err);
+    }
+    exec('npm install --global npm@2', function (err) {
+      if (err) {
+        throw new Error(err);
+      }
+      // I'm unsure what this one is for, but whatever.
+      exec('npm install --global npm', function (err) {
```

Comment:
> we don't need to actually upgrade to v3? so, we can skip that.

But maybe it wouldn't be a bad idea to test if dependencies install correctly while using the current version of npm.

> the shell script won't run on windows. this is premature optimization, because I know that we'll need to run IE, and I don't want to have anything in here that isn't portable. it also opens us up to running unit tests on diff platforms eventually.

Good call.